gammapy/estimators/flux_estimator.py

```diff
@@ -0,0 +1,161 @@
+
+class FluxEstimator(ParameterEstimator):
+    """Flux estimator.
+
+    Estimates flux for a given list of datasets with their model in a given energy range.
+
+    To estimate the model flux the amplitude of the reference spectral model is
+    fitted within the energy range. The amplitude is re-normalized using the "norm" parameter,
+    which specifies the deviation of the flux from the reference model in this
+    energy range.
+
+    Parameters
+    ----------
+    datasets : list of `~gammapy.spectrum.SpectrumDataset`
+        Spectrum datasets.
+    source : str or int
+        For which source in the model to compute the flux points.
+    norm_min : float
+        Minimum value for the norm used for the fit statistic profile evaluation.
+    norm_max : float
+        Maximum value for the norm used for the fit statistic profile evaluation.
+    norm_n_values : int
+        Number of norm values used for the fit statistic profile.
+    norm_values : `numpy.ndarray`
+        Array of norm values to be used for the fit statistic profile.
+    sigma : int
+        Sigma to use for asymmetric error computation.
+    sigma_ul : int
+        Sigma to use for upper limit computation.
+    reoptimize : bool
+        Re-optimize other free model parameters.
+    """
+
+    def __init__(
+            self,
+            datasets,
+            source=0,
+            norm_min=0.2,
+            norm_max=5,
+            norm_n_values=11,
+            norm_values=None,
+            sigma=1,
+            sigma_ul=3,
+            reoptimize=True,
+    ):
+        # make a copy to not modify the input datasets
+        datasets = self._check_datasets(datasets)
+
+        if not (datasets.is_all_same_type and datasets.is_all_same_shape):
+            raise ValueError(
+                "Flux point estimation requires a list of datasets"
+                " of the same type and data shape."
+            )
+
+        datasets = datasets.copy()
+
+        dataset = datasets[0]
+
+        model = dataset.models[source].spectral_model
+
+        self.model = ScaleSpectralModel(model)
+        self.model.norm.min = 0
+        self.model.norm.max = 1e5
+
+        if norm_values is None:
+            norm_values = np.logspace(
+                np.log10(norm_min), np.log10(norm_max), norm_n_values
+            )
+        self.norm_values = norm_values
+
+        self.source = source
+        datasets = self._set_scale_model(datasets)
+
+        super().__init__(
+            datasets,
+            sigma,
+            sigma_ul,
+            reoptimize,
+        )
+
+    @property
+    def ref_model(self):
+        return self.model.model
+
+    def __str__(self):
+        s = f"{self.__class__.__name__}:\n"
+        s += str(self.datasets) + "\n"
+        s += str(self.model) + "\n"
+        return s
+
+    def _set_scale_model(self, datasets):
+        # set the model on all datasets
+        for dataset in datasets:
+            if len(dataset.models) > 1:
```

Comment:
I think this `if...else` block can go just indexing into `models` should be fine.

Reply:
It's gone.